atlasdb-client/src/main/java/com/palantir/atlasdb/cache/DefaultOffHeapCache.java

```diff
@@ -0,0 +1,178 @@
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.LongSupplier;
+import java.util.stream.Collectors;
+
+import org.immutables.value.Value;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import com.google.common.util.concurrent.Futures;
+import com.palantir.atlasdb.autobatch.Autobatchers;
+import com.palantir.atlasdb.autobatch.CoalescingRequestFunction;
+import com.palantir.atlasdb.autobatch.DisruptorAutobatcher;
+import com.palantir.atlasdb.persistent.api.LogicalPersistentStore;
+import com.palantir.atlasdb.persistent.api.PhysicalPersistentStore.StoreNamespace;
+import com.palantir.common.streams.KeyedStream;
+import com.palantir.logsafe.exceptions.SafeIllegalArgumentException;
+import com.palantir.tritium.metrics.registry.MetricName;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+
+final class DefaultOffHeapCache<K, V> implements OffHeapCache<K, V> {
+    private static final Logger log = LoggerFactory.getLogger(DefaultOffHeapCache.class);
+    private static final String BATCHER_PURPOSE = "off-heap-cache";
+    private static final MetricName CACHE_HIT = constructCacheMetricName("cacheHit");
+    private static final MetricName CACHE_MISS = constructCacheMetricName("cacheMiss");
+    private static final MetricName CACHE_NUKE = constructCacheMetricName("cacheNuke");
+    private static final MetricName CACHE_SIZE = constructCacheMetricName("cacheSize");
+
+    private final LogicalPersistentStore<K, V> persistentStore;
+    private final LongSupplier maxSize;
+    private final AtomicReference<CacheDescriptor> cacheDescriptor = new AtomicReference<>();
+    private final TaggedMetricRegistry taggedMetricRegistry;
+    private final ConcurrentMap<K, V> inFlightRequests = new ConcurrentHashMap<>();
+    private final DisruptorAutobatcher<Map.Entry<K, V>, Map.Entry<K, V>> valuePutter;
+
+    DefaultOffHeapCache(
+            LogicalPersistentStore<K, V> persistentStore,
+            CacheDescriptor cacheDescriptor,
+            LongSupplier maxSize,
+            TaggedMetricRegistry taggedMetricRegistry) {
+        this.persistentStore = persistentStore;
+        this.cacheDescriptor.set(cacheDescriptor);
+        this.maxSize = maxSize;
+        this.taggedMetricRegistry = taggedMetricRegistry;
+        this.valuePutter = Autobatchers.coalescing(new WriteBatcher<>(this))
+                .safeLoggablePurpose(BATCHER_PURPOSE)
+                .build();
+        Gauge<Integer> cacheSizeGauge = () -> DefaultOffHeapCache.this.cacheDescriptor.get().currentSize().intValue();
```

Comment:
do you need to specify `DefaultOffHeapCache` out of curiosity? if no then get rid of it please.

Reply:
No, removed.